addons/project/project.py

```diff
@@ -62,15 +62,9 @@ class project_task_type(osv.osv):
                                'there are no records in that stage to display.'),
     }
 
-    def _get_default_project_ids(self, cr, uid, ctx={}):
-        project_id = self.pool['project.task']._get_default_project_id(cr, uid, context=ctx)
-        if project_id:
-            return [project_id]
-        return None
-
     _defaults = {
         'sequence': 1,
-        'project_ids': _get_default_project_ids,
+        'project_ids': lambda self, cr, uid, ctx: [ctx['default_project_id']] if ctx.get('default_project_id') else None
```

Comment:
Context can be None. Keep the method and put this code in the default + context check.